localstack/services/kinesisfirehose/resource_providers/aws_kinesisfirehose_deliverystream.py

```diff
@@ -0,0 +1,492 @@
+class SplunkDestinationConfiguration(TypedDict):
+    HECEndpoint: Optional[str]
+    HECEndpointType: Optional[str]
+    HECToken: Optional[str]
+    S3Configuration: Optional[S3DestinationConfiguration]
+    CloudWatchLoggingOptions: Optional[CloudWatchLoggingOptions]
+    HECAcknowledgmentTimeoutInSeconds: Optional[int]
+    ProcessingConfiguration: Optional[ProcessingConfiguration]
+    RetryOptions: Optional[SplunkRetryOptions]
+    S3BackupMode: Optional[str]
+
+
+class HttpEndpointConfiguration(TypedDict):
+    Url: Optional[str]
+    AccessKey: Optional[str]
+    Name: Optional[str]
+
+
+class HttpEndpointCommonAttribute(TypedDict):
+    AttributeName: Optional[str]
+    AttributeValue: Optional[str]
+
+
+class HttpEndpointRequestConfiguration(TypedDict):
+    CommonAttributes: Optional[list[HttpEndpointCommonAttribute]]
+    ContentEncoding: Optional[str]
+
+
+class HttpEndpointDestinationConfiguration(TypedDict):
+    EndpointConfiguration: Optional[HttpEndpointConfiguration]
+    S3Configuration: Optional[S3DestinationConfiguration]
+    BufferingHints: Optional[BufferingHints]
+    CloudWatchLoggingOptions: Optional[CloudWatchLoggingOptions]
+    ProcessingConfiguration: Optional[ProcessingConfiguration]
+    RequestConfiguration: Optional[HttpEndpointRequestConfiguration]
+    RetryOptions: Optional[RetryOptions]
+    RoleARN: Optional[str]
+    S3BackupMode: Optional[str]
+
+
+class Tag(TypedDict):
+    Key: Optional[str]
+    Value: Optional[str]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class KinesisFirehoseDeliveryStreamProvider(
+    ResourceProvider[KinesisFirehoseDeliveryStreamProperties]
+):
+
+    TYPE = "AWS::KinesisFirehose::DeliveryStream"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[KinesisFirehoseDeliveryStreamProperties],
+    ) -> ProgressEvent[KinesisFirehoseDeliveryStreamProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/DeliveryStreamName
+
+
+
+        Create-only properties:
+          - /properties/DeliveryStreamName
+          - /properties/DeliveryStreamType
+          - /properties/ElasticsearchDestinationConfiguration/VpcConfiguration
+          - /properties/AmazonopensearchserviceDestinationConfiguration/VpcConfiguration
+          - /properties/AmazonOpenSearchServerlessDestinationConfiguration/VpcConfiguration
+          - /properties/KinesisStreamSourceConfiguration
+
+        Read-only properties:
+          - /properties/Arn
+
+        IAM permissions required:
+          - firehose:CreateDeliveryStream
+          - firehose:DescribeDeliveryStream
+          - iam:GetRole
+          - iam:PassRole
+          - kms:CreateGrant
+          - kms:DescribeKey
+
+        """
+        model = request.desired_state
+        firehose = request.aws_client_factory.firehose
+        parameters = [
```

Comment:
The method `add_defaults` in the `FirehoseDeliveryStream` adds a default value for `DeliveryStreamName` if it's not supplied. Could you add this?

Reply:
Sure thing. I'll add it.

Reply:
Done